x-pack/plugins/index_management/public/application/sections/home/index_list/details_page/trained_models_deployment_modal.tsx

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { EuiConfirmModal, useGeneratedHtmlId, EuiHealth } from '@elastic/eui';
+import React from 'react';
+
+import { EuiLink } from '@elastic/eui';
+import { useEffect, useState } from 'react';
+import type { SharePluginStart } from '@kbn/share-plugin/public';
+import { i18n } from '@kbn/i18n';
+
+interface SemanticTextProps {
+  isSemanticTextEnabled: boolean;
+  setIsModalVisible: (isVisible: boolean) => void;
+  refreshModal: () => void;
+  pendingDeployments: Array<string | undefined>;
+  errorsInTrainedModelDeployment: string[];
+  url?: SharePluginStart['url'];
+}
+
+const ML_APP_LOCATOR = 'ML_APP_LOCATOR';
+const TRAINED_MODELS_MANAGE = 'trained_models';
+
+export function TrainedModelsDeploymentModal({
+  isSemanticTextEnabled,
+  setIsModalVisible,
+  refreshModal,
+  pendingDeployments = [],
+  errorsInTrainedModelDeployment = [],
+  url,
+}: SemanticTextProps) {
+  const modalTitleId = useGeneratedHtmlId();
+  const closeModal = () => setIsModalVisible(false);
+  const [mlManagementPageUrl, setMlManagementPageUrl] = useState<string>('');
+
+  useEffect(() => {
+    const mlLocator = url?.locators.get(ML_APP_LOCATOR);
+    const generateUrl = async () => {
+      if (mlLocator) {
+        const mlURL = await mlLocator.getUrl({
+          page: TRAINED_MODELS_MANAGE,
+        });
+        setMlManagementPageUrl(mlURL);
```

Comment:
Nit: recommendation to handle case where modal is unmounted before `setMlManagementPageUrl` is called. Something similar to this or using `useMountedState` from `react-use/lib/useMountedState`:

```
  useEffect(() => {
    let isCancelled = false;
    const generateUrl = async (_params: MlLocatorParams) => {
      if (mlLocator) {
        const url = await mlLocator.getUrl(_params, getUrlParams);
        if (!isCancelled) {
          setHref(url);
        }
      }
    };
    generateUrl(params);
    return () => {
      isCancelled = true;
    };
  }, [params, getUrlParams]);
```

Reply:
Updated